update_arrow_test.c

```diff
@@ -0,0 +1,32 @@
+/**
+* @file update_arrow_test.c
+* @author Jaser
+*/
+
+#include "map.h"
+#include <assert.h>
+
+void update_arrow_test()
+{
+    map_t map = {0};
+    arrow_t arrow[MAP_MAX_NUM_OF_ARROWS] = {
+      {{.x = 30,.y = 30} ,5,DIRECTION_UP},
+      {{.x = 3,.y = 90} ,5,DIRECTION_DOWN},
+      {{.x = 10,.y = 0} ,5,DIRECTION_UP},
+      {{.x = 10,.y = 100} ,5,DIRECTION_DOWN},
+    };
+
+    map.space.x_min = 0;
+    map.space.x_max = 100; 
+    map.space.y_min = 0;
+    map.space.y_max = 100;
+    map.number_of_arrows = 10;
```

Comment:
If you declare that the number of arrows are 10 , you should _initialize_ them all. In this case you should either declare other arrows or change number_of_arrows to 4.